develop/tutorials/articles/300-osgi-basics/03-adding-third-party-libraries-to-a-module.markdown

```diff
@@ -1,51 +1,39 @@
-# Adding Third Party Libraries to a Module [](id=adding-third-party-libraries-to-a-module)
+# Resolving Third Party Library Package Dependencies [](id=adding-third-party-libraries-to-a-module)
 
-The OSGi framework is intended for modular development where you can create
-modules containing explicit information about their requirements and
-capabilities that can be used to assemble a working system. In a perfect world,
-every library would contain this information (i.e., as OSGi modules).
-Unfortunately, not all libraries are packaged this way. So how do you use
-them in OSGi?
+The OSGi framework lets you build applications composed of multiple modules. The
+modules must resolve their Java package dependencies for the framework to
+assemble the modules into a working system. In a perfect world, every Java
+library would be an OSGi bundle (module), but many libraries aren't. So how do
+you resolve the packages your OSGi module needs from non-OSGi third party
+libraries?
 
-You can add them within your module. This isn't the only way, but it's the
-method that most resembles building WAR files, so it's the one we recommend for
-beginners.
+Here is the main workflow for resolving third party Java library packages:
 
-There are a couple different ways to add libraries to your modules:
+**Step 1 - Find an OSGi module version of the library**: Repositories, such as
+Eclipse Orbit, host OSGi modules that developers have created based on
+traditional Java libraries.
+[Eclipse Orbit](download.eclipse.org/tools/orbit/downloads/drops/R20170919201930/)
+hosts over a hundred such OSGi modules. Deploying the module to Liferay's OSGi
+framework lets you share it on the system. If you find a module for the library
+you need, use it. Otherwise, go to Step 2. 
```

Comment:
Lets mention what it means to `use it`.  It could be something like `add the jar to osgi/modules` folder or `deploy` folder to your bundle and start it.  Then in your own module you simply have a `compileOnly` dependency and everything will be resolved correctly when you deploy your own module.